### PR TITLE
kotlin-language-server: fix missing JDK in PATH

### PR DESCRIPTION
**Pull request by vtuan10**
###### Description of changes
The `kotlin-language-server` crashes on start. Fixed by adding the `JDK` to `PATH` and set `JAVA_HOME`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veehaitch** [comment]
Package lacks a maintainer. Feel free to add yourself if that's okay for you 🙂

File: pkgs/development/tools/kotlin-language-server/default.nix [hunk 1]
```diff
@@ -18,13 +18,18 @@ stdenv.mkDerivation rec {
...
+  postFixup = ''
+    wrapProgram "$out/bin/kotlin-language-server" --set JAVA_HOME ${openjdk11} --prefix PATH : ${lib.strings.makeBinPath [ openjdk11 maven ] }
+  '';
 
   meta = {
     description = "kotlin language server";
     longDescription = ''
       About Kotlin code completion, linting and more for any editor/IDE
       using the Language Server Protocol Topics'';
+    maintainers = with maintainers; [ vtuan10 ];
```
Comment (veehaitch) [inline pkgs/development/tools/kotlin-language-server/default.nix:32]:
```suggestion
    maintainers = with lib.maintainers; [ vtuan10 ];
```

File: pkgs/development/tools/kotlin-language-server/default.nix [hunk 2]
```diff
@@ -18,13 +18,18 @@ stdenv.mkDerivation rec {
   '';
 
   nativeBuildInputs = [ unzip gradle ];
-  buildInputs = [ openjdk11 gradle ];
+  buildInputs = [ openjdk11 gradle makeWrapper ];
```
[the remark this answers is not in the document]
Comment (vtuan10) [reply pkgs/development/tools/kotlin-language-server/default.nix]: Done.

File: pkgs/development/tools/kotlin-language-server/default.nix [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
 
 stdenv.mkDerivation rec {
   pname = "kotlin-language-server";
-  version = "1.2.0";
-  src = fetchurl {
+  version = "1.3.0";
+  src = fetchzip {
     url = "https://github.com/fwcd/kotlin-language-server/releases/download/${version}/server.zip";
-    sha256 = "sha256-GXr+sOGa09qeQrfaJYRPsoHULwfMst8tTr4y2cv752k=";
+    sha256 = "sha256-nqrZoSntomNLUMTni/yoiVhDOJFSFmZeGze8IJhITu0=";
```
Comment (veehaitch) [inline pkgs/development/tools/kotlin-language-server/default.nix:8]: Nit

```suggestion
    hash = "sha256-nqrZoSntomNLUMTni/yoiVhDOJFSFmZeGze8IJhITu0=";
```

**Comment by ajs124** [comment]
Looks like it needs a rebase

**Comment by vtuan10** [comment]
> Looks like this needs a rebase

Ah, it was updated. Fixed.

**Pull request merged**